ykrt/src/location.rs

```diff
@@ -440,10 +440,10 @@ pub(crate) struct HotLocation {
 pub(crate) enum HotLocationKind {
     /// Points to executable machine code that can be executed instead of the interpreter for this
     /// HotLocation.
-    Compiled(*const CompiledTrace),
+    Compiled(Arc<CompiledTrace>),
```

Comment:
OK, so perhaps a silly question, but why `Arc` at all? Why not just `CompiledTrace`? Is there some lifetime and concurrency stuff going on here?

Reply:
At the moment, we could get away with `Box<CompiledTrace>`. However, for side traces, we will have to keep a reference to the beginning of the loop -- so `CompiledTrace` will have to be shared somehow, probably via an `Arc`. *Where* that `Arc` goes is an open question: it might end up inside `CompiledTrace`. So it seems better to bite the bullet and make that sharing explicit now.

Reply:
OK, let's roll with this for now.